ERCS/erc-7588.md

```diff
@@ -0,0 +1,102 @@
+---
+eip: 7588
+title: Blob Transactions Metadata JSON Schema
+description: Attaching metadata to blobs carried by blob transactions
+author: Gavin Fu (@gavfu), Leo Wang (@wanglie1986), Aiden X (@4ever9), Bova Chen (@appoipp)
+discussions-to: https://ethereum-magicians.org/t/erc7588-attaching-metadata-to-blobs-carried-by-blob-transactions/17873
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-01-01
+requires: 4844
+---
+
+## Abstract
+
+This EIP proposes a standard for attaching metadata to blobs carried by shard blob transactions, as introduced by [EIP-4844](./eip-4844.md). The metadata is a JSON object that conforms to a predefined schema and is encoded in hexadecimal with UTF-8 character encoding. The metadata is put in the `data` field of the blob transaction.
+
+## Motivation
+
+[EIP-4844](./eip-4844.md) defines a new type of transaction, called “blob transaction”, which contains a list of blobs and their KZG commitments and proofs. Blob transactions are designed to be used by rollups to post their layer 2 transaction data to Ethereum layer 1.
+
+Typically, rollups keep track of their own posted blob transactions, download blobs and decode their layer 2 transaction data from blobs on demand. However, some third party solutions, such as Portal Network, blobscan, etc., may also index all the blobs ever posted to Ethereum and provide querying services of them. It would greatly improve the visibility and auditability of blobs if we could attach some metadata to the blobs, such as `originator`, `description`, `content_type`, etc.
+
+Moreover, some `dStorage` projects may use blob transactions to post user data to Ethereum, sync and permanently store the blobs off-chain for future retrieval. Attaching metadata to blobs would also inspire novel applications, such as inscriptions, etc.
+
+
+## Specification
+
+### Metadata JSON Schema
+
+The metadata is a JSON object that conforms to the following schema:
```

Comment:
It'd be nice to specify what format the schema is in (I would guess JSON Schema?)